podman_tutorial.md

```diff
@@ -0,0 +1,153 @@
+# cd ~/src/github.com/projectatomic/
+# git clone https://github.com/projectatomic/libpod/
+#
+# cd libpod
+# make
+# sudo make install PREFIX=/usr
+# sudo mkdir -p /etc/containers
+# sudo bash -c 'cat <<EOF > /etc/containers/registries.conf
+registries.search
+registries = ['docker.io', 'registry.fedoraproject.org']
+EOF
+'
+# sudo bash -c 'cat <<EOF > /etc/containers/policy.json
+ {
+     "default": [
+         {
+             "type": "insecureAcceptAnything"
+         }
+     ],
+     "transports":
+         {
+             "docker-daemon":
+                 {
+                     "": [{"type":"insecureAcceptAnything"}]
+                 }
+         }
+ }
+ EOF
+'
+# git clone https://github.com/containernetworking/plugins.git ~/src/github.com/containernetworking/plugins
+# cd ~/src/github.com/containernetworking/plugins
+# ./build
+# sudo mkdir -p /usr/libexec/cni
+# sudo cp bin/* /usr/libexec/cni
+# git clone https://github.com/opencontainers/runc.git ~/src/github.com/opencontainers/runc
+# GOPATH=~/ make static BUILDTAGS="seccomp selinux"
+# sudo cp runc /usr/bin/runc
+# sudo mkdir -p /usr/local/libexec
+# sudo ln -s /usr/libexec/crio /usr/local/libexec/
+```
+
+) -->
+## Familiarizing yourself with podman
+
+### Running a sample container
+This sample container will run a very basic httpd server that serves only its index
+page.
+```
+# sudo podman run -dt -e HTTPD_VAR_RUN=/var/run/httpd -e HTTPD_MAIN_CONF_D_PATH=/etc/httpd/conf.d \
+                    -e HTTPD_MAIN_CONF_PATH=/etc/httpd/conf \
+                    -e HTTPD_CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/httpd/ \
+                    registry.fedoraproject.org/f26/httpd /usr/bin/run-httpd
+```
+Note the Container ID returned by that command.
+
+### Listing running containers
+Once you have created or run a container, you can list the known containers with the *ps* command.
+```
+# sudo podman ps
+```
+
+Note: If you add *-a* to the *ps* command, it podman will show all containers.
+
+### Executing a command in a running container
+You can use the *exec* subcommand to execute a command in a running container.  In this case, we
+need to install iproute in the container so we can obtain its IP address.
+```
+# sudo podman exec -t <container_id> dnf install iproute
+# sudo podman exec -t <container_id> ip a
+```
+
+Note the IP address of the *eth0* device.
+
+### Testing the httpd server
+Now that we have the IP address of the container, we can test the network communication between the host
+operating system and the container using curl. The following command should display the index page of our
+containerized httpd server.
+```
+# curl http://<IP_address>:8080
+```
+
+### Viewing the container's logs
+You can view the container's logs with podman as well:
+```
+# sudo podman logs <container_id>
+```
+
+### Viewing the container's pids
+And you can observe the httpd pid in the container with *top*.
+```
+# sudo podman top <container_id>
```

Comment:
This crashed and burned.  If this is unexpected please spin up an issue.

```
sudo podman top 46749a3c05d308a269649d1a014137eb53aa110226ba17e7443dc8cab12ae3ce
panic: runtime error: index out of range

goroutine 1 [running]:
main.psDataToPSParams(0xc420070310, 0xb, 0xb, 0xc42006e800, 0x8, 0x8, 0xc, 0x0, 0x0, 0x221d0de, ...)
	/root/src/github.com/projectatomic/libpod/_output/src/github.com/projectatomic/libpod/cmd/podman/top.go:151 +0x46e
main.topCmd(0xc420154dc0, 0x0, 0x0)
	/root/src/github.com/projectatomic/libpod/_output/src/github.com/projectatomic/libpod/cmd/podman/top.go:87 +0x3d9
github.com/projectatomic/libpod/vendor/github.com/urfave/cli.HandleAction(0x1ef65e0, 0x22cc488, 0xc420154dc0, 0x0, 0xc420387080)
	/root/src/github.com/projectatomic/libpod/_output/src/github.com/projectatomic/libpod/vendor/github.com/urfave/cli/app.go:501 +0xd2
github.com/projectatomic/libpod/vendor/github.com/urfave/cli.Command.Run(0x2218892, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2254845, 0x2c, 0x0, ...)
	/root/src/github.com/projectatomic/libpod/_output/src/github.com/projectatomic/libpod/vendor/github.com/urfave/cli/command.go:165 +0x505
github.com/projectatomic/libpod/vendor/github.com/urfave/cli.(*App).Run(0xc420494a80, 0xc420010090, 0x3, 0x3, 0x0, 0x0)
	/root/src/github.com/projectatomic/libpod/_output/src/github.com/projectatomic/libpod/vendor/github.com/urfave/cli/app.go:259 +0x740
main.main()
	/root/src/github.com/projectatomic/libpod/_output/src/github.com/projectatomic/libpod/cmd/podman/main.go:155 +0x1112
```

Reply:
yeah, thats https://github.com/projectatomic/libpod/issues/169

Reply:
commented out for now